test/test_autograd.py

```diff
@@ -4192,6 +4192,40 @@ def test_power_function(self):
         c.backward()
         self.assertEqual(b.grad, torch.tensor([-inf, 0., 0.]))
 
+    def test_nanprod_with_nans(self):
+        a = torch.randn(2, 2, 2, 2)
+        with torch.no_grad():
+            a[a < 0.2] = float('nan')
+        a.requires_grad = True
```

Comment:
nit: you can actually move this as a kwargs of the randn above since you use no_grad() to do the modification.

Reply:
This makes sense, thanks @albanD!